From a41966a2056b6bc873b7314457c4c965f1b7d59b Mon Sep 17 00:00:00 2001
From: Li Zhou <li.zhou@windriver.com>
Date: Mon, 28 Jun 2021 11:37:20 +0800
Subject: [PATCH 3/3] Add mcx16 cflag to solve linking issue

Linking error is reported as:
  MODPOST .../quickassist/lookaside/access_layer/src/Module.symvers
ERROR: modpost: "__sync_bool_compare_and_swap_16"
[.../quickassist/lookaside/access_layer/src/qat_api.ko] undefined!

Add -mcx16 option to enables GCC to generate CMPXCHG16B instructions
in 64-bit code to implement compare-and-exchange operations on 16-byte
aligned 128-bit objects.

Signed-off-by: Li Zhou <li.zhou@windriver.com>
---
 quickassist/lookaside/access_layer/src/common/utils/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/quickassist/lookaside/access_layer/src/common/utils/Makefile b/quickassist/lookaside/access_layer/src/common/utils/Makefile
index 1f24380..6948f8c 100644
--- a/quickassist/lookaside/access_layer/src/common/utils/Makefile
+++ b/quickassist/lookaside/access_layer/src/common/utils/Makefile
@@ -112,6 +112,8 @@ ifdef KPT
 EXTRA_CFLAGS+=-DKPT
 endif
 
+EXTRA_CFLAGS+=-mcx16
+
 #common includes between all supported OSes
 INCLUDES+=-I$(LAC_DIR)/src/common/include\
           -I$(LAC_DIR)/src/common/compression/include\
-- 
2.17.1

